org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/WRecordIDDialog.java

```diff
@@ -0,0 +1,196 @@
+* This program is free software; you can redistribute it and/or       *
+* modify it under the terms of the GNU General Public License         *
+* as published by the Free Software Foundation; either version 2      *
+* of the License, or (at your option) any later version.              *
+*                                                                     *
+* This program is distributed in the hope that it will be useful,     *
+* but WITHOUT ANY WARRANTY; without even the implied warranty of      *
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the        *
+* GNU General Public License for more details.                        *
+*                                                                     *
+* You should have received a copy of the GNU General Public License   *
+* along with this program; if not, write to the Free Software         *
+* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,          *
+* MA 02110-1301, USA.                                                 *
+*                                                                     *
+* Contributors:                                                       *
+* - Peter Takacs, Cloudempiere                                        *
+**********************************************************************/
+package org.adempiere.webui.window;
+
+import org.adempiere.webui.component.Button;
+import org.adempiere.webui.component.ConfirmPanel;
+import org.adempiere.webui.component.Textbox;
+import org.adempiere.webui.component.Window;
+import org.adempiere.webui.editor.WRecordIDEditor;
+import org.adempiere.webui.editor.WSearchEditor;
+import org.adempiere.webui.editor.WTableDirEditor;
+import org.adempiere.webui.factory.ButtonFactory;
+import org.compiere.model.GridField;
+import org.compiere.model.MColumn;
+import org.compiere.model.MLookup;
+import org.compiere.model.MLookupFactory;
+import org.compiere.model.MLookupInfo;
+import org.compiere.model.MTable;
+import org.compiere.util.DisplayType;
+import org.compiere.util.Env;
+import org.compiere.util.Msg;
+import org.zkoss.zhtml.Text;
+import org.zkoss.zk.ui.Page;
+import org.zkoss.zk.ui.event.Event;
+import org.zkoss.zk.ui.event.EventListener;
+import org.zkoss.zk.ui.event.Events;
+import org.zkoss.zul.Div;
+import org.zkoss.zul.Vlayout;
+
+/**
+ * 
+ * @author Peter Takacs, Cloudempiere
+ *
+ */
+public class WRecordIDDialog extends Window implements EventListener<Event> {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 8126107952514403099L;
+	
+	private Page page;
+	private WRecordIDEditor editor;
+	private GridField tableIDGridField;
+	
+	private Div contentDiv;
+	private Div labelsDiv;
+	private Div fieldsDiv;
+	private Div confirmPanelDiv;
+	private WTableDirEditor tableIDEditorCopy;
+	
+	private Button okBtn;
+	private Button cancelBtn;
+	private Textbox parentTextBox;
+	private WSearchEditor recordsEditor;
+	
+	private Integer value;
```

Comment:
you are really managing 2 instead of 1 value here. more precise to have tableIDValue and recordIDValue instead of just value.